includes/checkout/template.php

```diff
@@ -972,15 +972,21 @@ function edd_checkout_button_next() {
  * @return string
  */
 function edd_checkout_button_purchase() {
-	$color = edd_get_option( 'checkout_color', 'blue' );
-	$color = ( $color == 'inherit' ) ? '' : $color;
-	$style = edd_get_option( 'button_style', 'button' );
-	$label = edd_get_checkout_button_purchase_label();
+	$enabled_gateways = edd_get_enabled_payment_gateways();
 
 	ob_start();
-?>
-	<input type="submit" class="edd-submit <?php echo $color; ?> <?php echo $style; ?>" id="edd-purchase-button" name="edd-purchase" value="<?php echo $label; ?>"/>
-<?php
+
+	if ( ! empty( $enabled_gateways ) || empty( edd_get_cart_total() ) ) {
```

Comment:
This will fatal on PHP 5.3 and earlier.

Reply:
Yep, fixing, Been writing too much EDD 3.0 code apparently.